File: server/app/routes/runs.py
from fastapi import APIRouter, HTTPException

from app.store.runs import get_run

router = APIRouter(prefix="/api/runs", tags=["runs"])


@router.get("/{run_id}")
def get_run_timeline(run_id: str):
    run = get_run(run_id)
    if not run:
        raise HTTPException(404, "run not found")
    return run

File: server/src/routes/runs.ts
import { Router } from "express";
import { getRun } from "../store/runs.js";

export const runsRouter = Router();

runsRouter.get("/:id", async (req, res) => {
  const run = await getRun(req.params.id);
  if (!run) {
    res.status(404).json({ error: "run not found" });
    return;
  }
  res.json(run);
});
